### PR TITLE
Fix broken hyperlinks and redirects prior to 0.9.0 release

### PR DESCRIPTION
**Pull request by namurphy**
I did a `make linkcheck` in the docs to find and fix broken hyperlinks and redirects, along with a few minor/incidental changes.

### CODE REVIEW

File: docs/contributing/coding_guide.rst [hunk 1]
```diff
@@ -268,7 +268,7 @@ the code is supposed to be doing.
 
 .. tip::
 
-   Most `integrated development environments <IDE>`_ (IDEs) have a
+   Most `integrated development environments <ide>`_ (IDEs) have a
```
Comment (namurphy) [inline docs/contributing/coding_guide.rst:271]: - [ ] Use `:wikipedia:` role instead?

File: docs/notebooks/analysis/nullpoint.ipynb [hunk 2]
```diff
@@ -53,7 +53,8 @@
    "id": "60702e11-905b-43b3-85ab-dc3f85bc9bab",
    "metadata": {},
    "source": [
-    "## How Null Point Finder works\n",
+    "## How the null point finder works\n",
+    "\n",
```
Comment (namurphy) [inline docs/notebooks/analysis/nullpoint.ipynb:57]: I made a few small changes in this notebook like adding blank lines in some Markdown cells.

File: docs/notebooks/analysis/nullpoint.ipynb [hunk 3]
```diff
@@ -198,7 +204,8 @@
    "mimetype": "text/x-python",
    "name": "python",
    "nbconvert_exporter": "python",
-   "pygments_lexer": "ipython3"
+   "pygments_lexer": "ipython3",
+   "version": "3.10.4"
```
Comment (namurphy) [inline docs/notebooks/analysis/nullpoint.ipynb:208]:
```suggestion
   "pygments_lexer": "ipython3"
```

File: docs/contributing/testing_guide.rst [hunk 4]
```diff
@@ -284,17 +284,16 @@ Using an integrated development environment
 -------------------------------------------
 
 Most IDEs have built-in tools that simplify software testing. IDEs like
-PyCharm_, `Visual Studio`_, and Atom_ allow test configurations to be
-run with a click of the mouse or a few keystrokes. While IDEs require
-time to learn, they are among the most efficient methods to
-interactively perform tests. Here are instructions for running tests in
-several popular IDEs:
+PyCharm_ and `Visual Studio`_ allow test configurations to be run with a
```
Comment (StanczakDominik) [inline docs/contributing/testing_guide.rst:287]: [*] for Atom!

**Review by StanczakDominik — APPROVED** [review]

**Pull request merged**